Wikipedia/Code/SectionEditorViewController.swift

```diff
@@ -95,19 +97,30 @@ class SectionEditorViewController: ViewController {
         navigationItemController.delegate = self
         
         loadEditNotices()
-        loadWikitext { [weak self] blockedError in
+        loadWikitext { [weak self] resultError in
             
             guard let self else {
                 return
             }
             
-            if let blockedError {
-                self.presentErrorMessage(blockedError: blockedError)
-                
-                DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.2) { // helps prevent flash as wikitext is loaded
-                    self.configureWebView(readOnly: true)
+            if let error = resultError {
+                if error.code.contains("protectedpage") {
+                    self.presentProtectedPageWarning(error: error)
+                    DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.2) { // helps prevent flash as wikitext is loaded
+
+                        self.configureWebView(readOnly: true)
```

Comment:
Looking at the task description, it seems like there's a chance that the user could have `canUserEdit = true` (meaning they have elevated permissions to edit a semi-protected page) and need an enabled editor, but design still expects the modal to present informing them of protection. (see "Editor decides to edit a semi-protected page" section in the Phab task).

So I think it may make more sense to combine the checks into one conditional section:
```
if {we have an error with a code of type protectedpage}  {
    {present warning}
    self.configureWebView(readOnly: !self.canUserEdit)
}
```

Reply:
Actually, this advice may have been wrong (I should have tested!) 😅 I don't think it's possible to get into a situation where the editor is enabled and a page protection error displays, because it seems like if the user has the right permissions, we don't receive any error text from the API for the modal.

No need to change anything though, this logic is all working fine. I may just leave a comment on the Phab task as an FYI.